plugins/block-shareable-procedures/test/index.js

```diff
@@ -3,34 +3,105 @@
  * Copyright 2022 Google LLC
  * SPDX-License-Identifier: Apache-2.0
  */
-
 /**
  * @fileoverview Block test.
  */
 
 import * as Blockly from 'blockly';
-import {createPlayground, toolboxCategories} from '@blockly/dev-tools';
-import {blocks, unregisterProcedureBlocks} from '../src/index';
+import {toolboxCategories} from '@blockly/dev-tools';
+import {blocks, unregisterProcedureBlocks, ObservableProcedureModel, ObservableParameterModel} from '../src/index';
+import {ProcedureBase} from '../src/events_procedure_base';
 
 
 unregisterProcedureBlocks();
 Blockly.common.defineBlocks(blocks);
 
-/**
- * Create a workspace.
- * @param {HTMLElement} blocklyDiv The blockly container div.
- * @param {!Blockly.BlocklyOptions} options The Blockly options.
- * @returns {!Blockly.WorkspaceSvg} The created workspace.
- */
-function createWorkspace(blocklyDiv, options) {
-  const workspace = Blockly.inject(blocklyDiv, options);
-  return workspace;
-}
+Blockly.serialization.registry.unregister('procedures');
+Blockly.serialization.registry.register(
+    'procedures',
+    new Blockly.serialization.procedures.ProcedureSerializer(
+        ObservableProcedureModel, ObservableParameterModel));
+
+export let workspace1;
+export let eventSharer1;
+export let workspace2;
+export let eventSharer2;
 
 document.addEventListener('DOMContentLoaded', function() {
-  const defaultOptions = {
+  const options = {
     toolbox: toolboxCategories,
   };
-  createPlayground(document.getElementById('root'), createWorkspace,
-      defaultOptions);
+  workspace1 = Blockly.inject('blockly1', options);
+  workspace2 = Blockly.inject('blockly2', options);
+  eventSharer1 = workspace1.addChangeListener(createEventSharer(workspace2));
+  eventSharer2 = workspace2.addChangeListener(createEventSharer(workspace1));
+
+  workspace1.addChangeListener(
+      createSerializationListener(workspace1, 'save1'));
+  workspace2.addChangeListener(
+      createSerializationListener(workspace2, 'save2'));
+
+  document.getElementById('load')
+      .addEventListener('click', () => reloadWorkspaces());
 });
+
+/**
+ * Returns an event listener that shares procedure and var events from the
+ * connected workspace to the other workspace.
+ * @param {!Blockly.Workspace} otherWorkspace The workspace to share events to.
+ * @returns {function(Blockly.Events.Abstract)} The listener.
+ */
+function createEventSharer(otherWorkspace) {
+  return (e) => {
+    if (!(e instanceof ProcedureBase) &&
+        !(e instanceof Blockly.Events.VarBase)) {
+      return;
+    }
+    let event;
+    try {
+      event = Blockly.Events.fromJson(e.toJson(), otherWorkspace);
+    } catch (e) {
+      console.log(
+          'Could not deserialize event. This is expected to happen. E.g. ' +
+          'when round-tripping parameter deletes, the delete in the ' +
```

Comment:
If this is expected to happen, we probably shouldn't log it in the console. Just a comment here would be better, I think.